openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -340,3 +340,10 @@ def pre_requisite_courses(self):
         Returns a list of ID strings for this course's prerequisite courses.
         """
         return json.loads(self._pre_requisite_courses_json)
+
+    @classmethod
+    def get_all_course_keys(cls):
+        """
+        Returns all course keys from course overviews.
+        """
+        return [course_overview.id for course_overview in cls.objects.all()]
```

Comment:
That model has a lot of fields in it, and this will pull the full rows from the database into python. You can use [`.values('id')`](https://docs.djangoproject.com/en/1.4/ref/models/querysets/#values) to limit it to just IDs (you'll have to change code a little -- `.values()` returns dicts)